### PR TITLE
Blog post about Agile Worksop on 3/12/15 - "Don't Go Chasing Waterfalls"

### PR DESCRIPTION
**Pull request by SteveWells**
My first blog post :-) 

Let me know I've completely messed it up...

### CODE REVIEW

File: internal/_posts/2015-12-03-dont-go-chasing-waterfalls.markdown [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+Although the argument in this blog is based around Scrum, the reasoning applies equally well to any Agile framework or process.
+
+## What Is A Mini-Waterfall
+In the Scrum Guide, only 3 roles are recognised in a Scrum team:
+
+* Product Owner
+* Scrum Master
+* Developers
+
+The idea being that the whole team take responsibility for all the work needed to get the software increments delivered. Software is produced collaboratively with the whole taking responsibility for getting the job done. Some developers may have "specialisations" - infrastructure, testing/QA, etc. - but all are developers are far as Scrum is concerned. In many teams (*my current team included*), however, other roles have crept in, so the team looks more like this:
```
Comment (craggy2593) [inline internal/_posts/2015-12-03-dont-go-chasing-waterfalls.markdown]: "all are developers are far as Scrum is concerned" - Doesn't make sense to me?

File: internal/_posts/2015-12-03-dont-go-chasing-waterfalls.markdown [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+### The team is not equally responsible
+When there are dedicated, non-developer roles in a team, the team is no longer collectively responsible for the quality of the product. Developers develop, then hand responsibility to the testers. This kind of "over-the-wall" thinking means that the team as a whole is no longer collectively responsible - developers won't look at bugs that they deem un-reproducible, and testers will complain that developers haven't done enough basic testing before handing over the code. 
+
+I have seen this in many teams.
+
+### Waterfall stops failing early
+A guiding principle of Agile is "fail early"; in fact, as early as possible. Don't wait until code has been written and handed over to a test team before bugs are discovered. Bugs should be identified as soon as code is checked in (*or earlier, if possible?*). Otherwise, you may be adding code that relies on code that already contains bugs. Very dangerous.
```
Comment (stephiepea) [inline internal/_posts/2015-12-03-dont-go-chasing-waterfalls.markdown]: "fail fast" is probably a more commonly used term, but up to you :)

**Comment by johnboyes** [comment]
:shipit:

File: internal/_posts/2015-12-03-dont-go-chasing-waterfalls.markdown (hunk 2 above)
Comment (SteveWells) [inline internal/_posts/2015-12-03-dont-go-chasing-waterfalls.markdown]: Yes, cheers for that....

**Pull request merged**